src/dispatch/stage.rs

```diff
@@ -242,20 +242,42 @@ impl<'a> StagesBuilder<'a> {
         let new_writes = new_writes.into_iter();
 
         let num_groups = ids[stage].len();
+        let mut dep_conflict = false;
 
         let conflict = (0..num_groups)
             .filter(|&group| {
                 let read_and_write = writes[stage][group]
                     .iter()
                     .chain(reads[stage][group].iter());
 
-                check_intersection(new_writes.clone(), read_and_write) ||
-                check_intersection(new_reads.clone(), writes[stage][group].iter()) ||
-                check_intersection(new_dep.iter(), ids[stage][group].iter())
+                let inters = check_intersection(new_writes.clone(), read_and_write) ||
+                check_intersection(new_reads.clone(), writes[stage][group].iter());
+
+                if !inters {
```

Comment:
I would avoid using negation when you have if else.

Reply:
Makes sense, I used negation because the code was lazily evaluated before..